WIP: Add support for `mbind`, `get_mempolicy` and `set_mempolicy`

krh opened the pull request:
This adds support for the `mbind`, `set_mempolicy` and `get_mempolicy` NUMA syscalls.  The `get_mempolicy` syscall has a few different modes of operation, depending on the flags, which is demultiplexed into `get_mempolicy_node` and `get_mempolicy_next_node` for now.  There's a couple of other modes that writes into the variable length bit array, which aren't implemented for now.

sunfishcode commented on tests/numa/mbind.rs~:
```diff
@@ -0,0 +1,50 @@
+        rustix::fs::seek(&alt, rustix::fs::SeekFrom::Current(0)),
+        Ok(12)
+    );
+}
```
Are mbind.rs~ and main.rs~ editor backups?

krh replied:
Yes, oops, that's emacs backup files.

krh replied:
Yes, oops, that's emacs backup files.

sunfishcode reviewed (commented):
At a first glance, this change looks reasonable!

krh commented:
Pushed a new version: removed the emacs backup files, fixed `cargo fmt` and warnings about unused imports. Remaining errors look like they're all from missing the new symbols in `linux-raw-sys`.

krh commented:
This one: https://github.com/sunfishcode/linux-raw-sys/pull/96

sunfishcode commented:
If you'd like this PR to be included in the rustix 0.38 series, please use linux-raw-sys 0.4.x. I've now released 0.4.12 with https://github.com/sunfishcode/linux-raw-sys/pull/96. Another option is to open this PR in the [1.0-staging branch](https://github.com/bytecodealliance/rustix/tree/1.0-staging).

krh commented:
> If you'd like this PR to be included in the rustix 0.38 series, please use linux-raw-sys 0.4.x. I've now released 0.4.12 with [sunfishcode/linux-raw-sys#96](https://github.com/sunfishcode/linux-raw-sys/pull/96). Another option is to open this PR in the [1.0-staging branch](https://github.com/bytecodealliance/rustix/tree/1.0-staging).

Oh, ok, thanks, I'll target 0.4.12 then.

sunfishcode commented:
Rustix 1.0.0 is now released, and it uses the latest linux-raw-sys, so it's no longer necessary to backport to linux-raw-sys 0.4.
